Archipelago.HollowKnight/IC/ArchipelagoItem.cs

```diff
@@ -1,24 +1,42 @@
-﻿using ItemChanger;
+﻿using Archipelago.MultiClient.Net.Enums;
+using ItemChanger;
 using ItemChanger.Tags;
 using ItemChanger.UIDefs;
 
 namespace Archipelago.HollowKnight.IC
 {
     public class ArchipelagoItem : AbstractItem
     {
-        public ArchipelagoItem(string name)
+        public ArchipelagoItem(string name, string targetSlotName = null, ItemFlags itemFlags = 0)
```

Comment:
I like this, well done. It might be prudent to add explicit tags so people will understand. At the end of the desc add things like "(Progression)" or "(Trap)" so people understand the text is meant to change and what it means.

Reply:
I'm up for revisions to these later, but I was trying to keep the descriptions within HK's theme and flavor.

Also Github discarded my first comment here.